.github/workflows/build_workflow.yml

```diff
@@ -1,18 +1,33 @@
-name: GitHub Actions Demo
-run-name: ${{ github.actor }} is testing out GitHub Actions 🚀
-on: [push]
+name: Build workflow
+run-name: Build workflow
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - master
+  pull_request:
```

Comment:
since this is a  public repository i think pull_request_target might be a better trigger event from a security perspective